meeteval/viz/visualize.js

```diff
@@ -1489,6 +1491,45 @@ class CanvasPlot {
         }
     }
 
+    function drawRecordingAudioButton(container, rangeSelector, name, recording_file) {
+        let update_audio = function (){
+            // Note: Deleting and adding an audio with the same content doesn't
+            //       trigger a load. Hence, no optimization necessary.
+            audio_div.selectAll("*").remove();
+            let lower = rangeSelector.lower_input.node().value;
+            let upper = rangeSelector.upper_input.node().value;
+            let value = "[" + lower + ":" + upper + "]";
+            let path = server_path.node().value + "/" + file_path.node().value + "::" + value;
+            if ( parseFloat(lower) < parseFloat(upper) ){
+                audio_div.append("div").text(value);
+                audio_div.append("audio")
+                    .on("error", () => {audio_div.append("div").text("Issue while loading the audio")})
+                    .attr("controls", "true")
+                    .attr("src", path).text(path);
+            } else {
+                audio_div.append("div").text("Invalid range: " + lower + " - " + upper);
+            }
+        };
+
+        let pill = container.append("div").classed("pill", true);
+        let audio_tooltip = addTooltip(pill);
+
+        pill.append("span").text((name.trim() !== "") ? "🔊 " + name : "🔊");
+
+        let input = audio_tooltip.append("div").style("display", "flex");
+        let server_path = input.append("input").attr("type", "text").attr("placeholder", "e.g. http://localhost:7777").property("value", "http://localhost:7777");
```

Comment:
This server_path should also affect the segment audios. I just don't know how to pass it through

Reply:
In General, it would be good to have a settings menu, that does some caching (Across files).
But I don't know how to do that or if it is possible (We don't want server interactions).

Reply:
Hmm. I am not sure, how useful it is, when it is only kept for one URL.

Reply:
I guess it's not super useful, but it's the only way to persist data. Browsers are not made for local files...